src/nunit-gui/Model/TestModel.cs

```diff
@@ -187,7 +187,11 @@ public void OnStartup()
 
         public void Dispose()
         {
-            throw new NotImplementedException();
+            if (IsPackageLoaded)
+                UnloadTests();
+
+            if (Runner != null)
```

Comment:
In this case, I don't think so. It's common in our Dispose and some other cleanup methods to have a sequence of `if` statements each governing some part of the disposal. In this case, the first one can't be changed to use the `?.` operator, while the second one can. That would leave us with

```C#
if (IsPackageLoaded)
    UnloadTests();

Runner?.Dispose();
```

This breaks the pattern and requires a bit more thought to see that it's a sequence of conditionals.